### DISCUSSION TITLE
Is this a bug in DataGrid's use of CellStyleClasses?

### DISCUSSION THREAD

**Discussion by nasterhe** [AvaloniaUI/Avalonia#17790] [Q&A] [answered]
Normally, double-clicking on any area of the Cell will enter the editing state, but when the CellStyleClasses are set to the DataGridColumn, I can only enter the editing state by double-clicking on the text in the Cell。This leads users to think that there is a problem with our software.
Is there any solution? Thank you.

**Comment by maxkatz6** [comment]
> I can only enter the editing state by double-clicking on the text in the Cell

Sounds like there is missing background on the cell. Since you have custom cell style class, you probably should set any Background there. "Transparent" works too. Null backgrounds are invisible for the hit test.

**Reply by nasterhe** [reply]
@maxkatz6 

```
<DataGrid.Styles>
  <Style Selector="DataGridCell.Number" >
    <Setter Property="Background" Value="{Binding Number,Converter={StaticResource UserAttentionConverter}}" x:DataType="vmt:ValueHelper"/>
  </Style>
</DataGrid.Styles>

```

```
internal sealed class UserAttentionConverter : IValueConverter
{
    public object? Convert(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        if (value is decimal d && d == 0)
            return Brushes.Red;
        return null;
    }

    public object? ConvertBack(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        throw new NotImplementedException();
    }
}
```



My CellStyleClasses are designed to set backgrounds, is there anything that needs to be improved? Thanks a lot.

**Reply by maxkatz6** [reply, marked as answer]
You set null background for most of your cells, as I understand.
As I mentioned above, null background is not hit-testable. You need to set Transparent at least.

**Reply by nasterhe** [reply]
> You set null background for most of your cells, as I understand. As I mentioned above, null background is not hit-testable. You need to set Transparent at least.

Understood, thank you
